### PR TITLE
Make scripts MPI safe

### PR DESCRIPTION
**Pull request by dliptai**

### CODE REVIEW

**Comment by dliptai** [comment]
Seem OK with 1 task, but get an error when running with multiple
```
INFO: Running simulation for 32000 ms after steady state (6000 ms) with PID control
Loading network...
Loaded the network, loading the steady state
Assertion failed: file /Users/runner/work/1/s/src/nrniv/savstate.cpp, line 1069
1 /Users/dliptai/anaconda/envs/neuron2/bin/nrniv: fread(tqs_->tdeliver, sizeof(double), n, f) == n
1  near line 0
1  {initializer._initialize()}
                            ^
        1 SaveState[0].fread(...)
Assertion failed: (npss_ == i), function readnet, file /Users/runner/work/1/s/src/nrniv/savstate.cpp, line 1060.
Assertion failed: (npss_ == i), function readnet, file /Users/runner/work/1/s/src/nrniv/savstate.cpp, line 1060.
[DaveMacBookPro:74877] *** Process received signal ***
Assertion failed: (npss_ == i), function readnet, file /Users/runner/work/1/s/src/nrniv/savstate.cpp, line 1060.
[DaveMacBookPro:74878] *** Process received signal ***
[DaveMacBookPro:74878] Signal: Abort trap: 6 (6)
[DaveMacBookPro:74878] Signal code:  (0)
[DaveMacBookPro:74878] [ 0] [DaveMacBookPro:74877] Signal: Abort trap: 6 (6)
[DaveMacBookPro:74877] Signal code:  (0)
[DaveMacBookPro:74877] [ 0] [DaveMacBookPro:74875] *** Process received signal ***
[DaveMacBookPro:74875] Signal: Abort trap: 6 (6)
[DaveMacBookPro:74875] Signal code:  (0)
[DaveMacBookPro:74875] [ 0] 0   libsystem_platform.dylib            0x00007ff8016e8dfd _sigtramp + 29
[DaveMacBookPro:74875] [ 1] 0   libsystem_platform.dylib            0x00007ff8016e8dfd _sigtramp + 29
[DaveMacBookPro:74877] [ 1] 0   libsystem_platform.dylib            0x00007ff8016e8dfd _sigtramp + 29
[DaveMacBookPro:74878] [ 1] 0   ???                                 0x0000000000000004 0x0 + 4
[DaveMacBookPro:74875] [ 2] 0   ???                                 0x0000000000000004 0x0 + 4
[DaveMacBookPro:74877] [ 2] 0   ???                                 0x0000000000000004 0x0 + 4
[DaveMacBookPro:74878] [ 2] --------------------------------------------------------------------------
MPI_ABORT was invoked on rank 1 in communicator MPI_COMM_WORLD
with errorcode -1.
```

TLDR;
```
Assertion failed: file /Users/runner/work/1/s/src/nrniv/savstate.cpp, line 1069
1 /Users/dliptai/anaconda/envs/neuron2/bin/nrniv: fread(tqs_->tdeliver, sizeof(double), n, f) == n
1  near line 0
1  {initializer._initialize()}
                            ^
        1 SaveState[0].fread(...)
Assertion failed: (npss_ == i), function readnet, file /Users/runner/work/1/s/src/nrniv/savstate.cpp, line 1060.
```

**Comment by dliptai** [comment]
Different error if resuming with a different number of mpi tasks than what the steady state was run with 
```
INFO: Running simulation for 32000 ms after steady state (6000 ms) with PID control
Loading network...
Loaded the network, loading the steady state
Assertion failed: (nt == i && nc == memb_list[i].nodecount), function read, file /Users/runner/work/1/s/src/nrniv/savstate.cpp, line 835.
Assertion failed: (nt == i && nc == memb_list[i].nodecount), function read, file /Users/runner/work/1/s/src/nrniv/savstate.cpp, line 835.
[DaveMacBookPro:76419] *** Process received signal ***
[DaveMacBookPro:76420] *** Process received signal ***
[DaveMacBookPro:76420] Signal: Abort trap: 6 (6)
[DaveMacBookPro:76420] Signal code:  (0)
[DaveMacBookPro:76420] [ 0] 0   libsystem_platform.dylib            0x00007ff8016e8dfd _sigtramp + 29
[DaveMacBookPro:76420] [ 1] [DaveMacBookPro:76419] Signal: Abort trap: 6 (6)
[DaveMacBookPro:76419] Signal code:  (0)
[DaveMacBookPro:76419] [ 0] 0   libsystem_platform.dylib            0x00007ff8016e8dfd _sigtramp + 29
[DaveMacBookPro:76419] [ 1] 0   ???                                 0x00007ff7b235d4b0 0x0 + 140701823521968
```

TLDR;
```
Assertion failed: (nt == i && nc == memb_list[i].nodecount), function read, file /Users/runner/work/1/s/src/nrniv/savstate.cpp, line 835.
```

**Comment by dliptai** [comment]
Issue is with `SaveState`. 
> In a parallel simulation, each host [ParallelContext.id()](https://nrn.readthedocs.io/en/latest/python/modelspec/programmatic/network/parcon.html#ParallelContext.id) , should write an id specific file. Note that the set of files is at least [ParallelContext.nhost()]

See https://nrn.readthedocs.io/en/latest/python/simctrl/savstate.html#SaveState.fread
